lib/util.ts

```diff
@@ -40,3 +40,14 @@ export async function jsonOrError(url: string, options: HttpOptions = {}): Promi
 
     return await response.json()
 }
+
+export async function retryOnError<T>(fn: () => Promise<T>, times: number): Promise<T> {
+    try {
+        return await fn()
+    } catch (err) {
+        if (times === 0) {
+            throw err
+        }
+        return retryOnError(fn, times - 1)
```

Comment:
Should we add a small sleep here?
Retrying 3 times within 150ms (accounting for network latency) might be too short if the beacon has a 800ms latency that day.

Reply:
I've added an integration test where it runs for 5 rounds against testnet without a problem; I'd prefer we keep an eye on it (especially while it's a prerelease) and add sleeps if anybody has a problem rather than adding them preemptively

Reply:
lemme try like 100 rounds for science

Reply:
okay did 5mins of rounds on the 3sec testnet network and there wasn't a single retry so either it's fine or my system clock is out of whack (or something else)